src/vanilla/utils/atomWithObservable.ts

```diff
@@ -81,43 +77,51 @@ export function atomWithObservable<Data>(
       observable = itself
     }
 
-    let resolve: ((result: Result) => void) | undefined
-    const makePending = () =>
-      new Promise<Result>((r) => {
-        resolve = r
-      })
-    const initialResult: Result | Promise<Result> =
+    // TODO: replace with https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/withResolvers
+    // once CI Node version is bump to 22+
```

Comment:
It's not just about CI Node version. We support old Node version specified in package.json.